packages/frontEnd/src/views/categoryDetails/filters/FilterByCities/FilterByCitiesForm.tsx

```diff
@@ -32,6 +53,8 @@ export const FilterByCitiesForm = () => {
               id="san-francisco"
               className="form-checkbox rounded"
               type="checkbox"
+              onChange={(event) => onChange(event, "San Francisco")}
+              checked={currentCheckbox === "San Francisco"}
```

Comment:
It seems to me that the comparison must be dynamic, because the city will change with the number of ads.